modules/drivers/presto-jdbc/test/metabase/test/data/presto_jdbc.clj

```diff
@@ -0,0 +1,135 @@
+  (defmethod sql.tx/field-base-type->sql-type [:presto-jdbc base-type] [_ _] db-type))
+
+;; in the past, we had to manually update our Docker image and add a new catalog for every new dataset definition we
+;; added. That's insane. Just use the `test-data` catalog and put everything in that, and use
+;; `db-qualified-table-name` like everyone else.
+(def ^:private test-catalog-name "test_data")
+
+(def ^:private test-schema-name "default")
+
+(defn- dash->underscore [nm]
+  (str/replace nm #"-" "_"))
+
+(defmethod tx/dbdef->connection-details :presto-jdbc
+  [_ _ {:keys [database-name]}]
+  {:host               (tx/db-test-env-var-or-throw :presto-jdbc :host "localhost")
+   :port               (tx/db-test-env-var :presto-jdbc :port "8080")
+   :user               (tx/db-test-env-var-or-throw :presto-jdbc :user "metabase")
+   :additional-options (tx/db-test-env-var :presto-jdbc :additional-options nil)
+   :ssl                (tx/db-test-env-var :presto-jdbc :ssl "false")
+   :catalog            (dash->underscore database-name)
+   :schema             (tx/db-test-env-var :presto-jdbc :schema nil)})
+
+(defmethod execute/execute-sql! :presto-jdbc
+  [& args]
+  (apply execute/sequentially-execute-sql! args))
+
+(remove-method load-data/load-data! :presto-jdbc)
+
+(defn- load-data [dbdef tabledef]
+  ;; the JDBC driver statements fail with a cryptic status 500 error if there are too many
+  ;; parameters being set in a single statement; these numbers were arrived at empirically
+  (let [chunk-size (case (:table-name tabledef)
+                     "people" 30
+                     "reviews" 40
+                     "orders" 30
+                     "venues" 50
+                     "products" 50
+                     "cities" 50
+                     "sightings" 50
+                     "incidents" 50
+                     "checkins" 25
+                     100)
+        load-fn    (load-data/make-load-data-fn load-data/load-data-add-ids
+                     (partial load-data/load-data-chunked pmap chunk-size))]
+    (load-fn :presto-jdbc dbdef tabledef)))
+
+(defmethod load-data/load-data! :presto-jdbc
+  [_ dbdef tabledef]
+  (load-data dbdef tabledef))
+
+(defn- jdbc-spec->connection
+  "This is to work around some weird interplay between clojure.java.jdbc caching behavior of connections based on URL,
+  combined with the fact that the Presto driver apparently closes the connection when it closes a prepare statement.
+  Therefore, create a fresh connection from the DriverManager."
+  [jdbc-spec]
+  (DriverManager/getConnection (format "jdbc:%s:%s" (:subprotocol jdbc-spec) (:subname jdbc-spec))
+    (connection-pool/map->properties (select-keys jdbc-spec [:user :SSL]))))
+
+(defmethod load-data/do-insert! :presto-jdbc
+  [driver spec table-identifier row-or-rows]
+  (let [statements (ddl/insert-rows-ddl-statements driver table-identifier row-or-rows)]
+    (with-open [conn (jdbc-spec->connection spec)]
```

Comment:
How did this not trigger the reflection warning linter? You should add a `^java.sql.Connection` tag on `jdbc-spec->connection`. Otherwise `with-open` doesn't know that `conn` is an `AutoCloseable`

Reply:
Not sure why the reflection warner didn't complain, but I fixed them.